include/seqan3/alignment/pairwise/alignment_configurator.hpp

```diff
@@ -0,0 +1,189 @@
+        using kernel_t = std::function<result_t(first_seq_t const &, second_seq_t const &)>;
+
+
+        // ----------------------------------------------------------------------------
+        // Test some basic preconditions
+        // ----------------------------------------------------------------------------
+
+        using alignment_contract_t = alignment_contract<remove_cvref_t<sequences_t>, config_t>;
+
+        static_assert(alignment_contract_t::expects_tuple_like_value_type(),
+                      "Alignment configuration error: "
+                      "The value type of the sequence ranges must model the seqan3::detail::tuple_like_concept "
+                      "and must contain exactly 2 elements.");
+
+        static_assert(alignment_contract_t::expects_valid_scoring_scheme(),
+                      "Alignment configuration error: "
+                      "Either the scoring scheme was not configured or the given scoring scheme cannot be invoked with "
+                      "the value types of the passed sequences.");
+
+        // ----------------------------------------------------------------------------
+        // Unsupported configurations
+        // ----------------------------------------------------------------------------
+
+        if constexpr (config_t::template exists<align_cfg::band>())
+            throw std::domain_error{"Banded alignments are yet not supported."};
+
+        // ----------------------------------------------------------------------------
+        // Configure the algorithm
+        // ----------------------------------------------------------------------------
+
+        // Use default edit distance if gaps are not set.
+        auto const & gaps = cfg.template value_or<align_cfg::gap>(gap_scheme{gap_score{-1}});
+        auto const & scoring_scheme = get<align_cfg::scoring>(cfg).value;
+
+        // Check if edit distance can be used?
+        if (gaps.get_gap_open_score() == 0)
+        {
+            // TODO: Instead of relying on nucleotide scoring schemes we need to be able to determine the edit distance
+            //       option via the scheme.
+            if constexpr (is_type_specialisation_of_v<remove_cvref_t<decltype(scoring_scheme)>,
+                                                      nucleotide_scoring_scheme>)
+            {
+                if ((scoring_scheme.score('A'_dna15, 'A'_dna15) == 0) &&
+                    (scoring_scheme.score('A'_dna15, 'C'_dna15)) == -1)
+                    return configure_edit_distance<kernel_t>(cfg);
+            }
+        }
+
+        // ----------------------------------------------------------------------------
+        // Unsupported configurations
+        // ----------------------------------------------------------------------------
+
+        if constexpr (config_t::template exists<align_cfg::aligned_ends>())
+            throw std::domain_error{"Configuring the aligned ends is yet not supported."};
```

Comment:
Just for understanding:

Why not `static_assert`?
It's static constexpr, so it needs to be known at compile time, right?

In any case, would a throw in a constexpr context occur during compiling? (Never tested it 😁 )